src/jdk.management.jfr/share/classes/jdk/management/jfr/FlightRecorderMXBeanImpl.java

```diff
@@ -282,14 +282,14 @@ public void setRecordingOptions(long recording, Map<String, String> options) thr
         for (Map.Entry<String, String> entry : ops.entrySet()) {
             Object key = entry.getKey();
             Object value = entry.getValue();
-            if (!(key instanceof String)) {
-                throw new IllegalArgumentException("Option key must not be null, or other type than " + String.class);
```

Comment:
Why is the type check removed?

Reply:
Because it must be a String as we are iterating over Map.Entry<String, String>

Reply:
Only at compile time. A user could send data over the network using other types.

Reply:
Thanks for clarifying this. I have reverted accordingly.